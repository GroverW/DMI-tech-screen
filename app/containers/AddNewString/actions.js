/*
 *
 * AddNewString actions
 *
 */

import { ADD_STRING, ADD_STRING_SUCCESS, ADD_STRING_ERROR } from './constants';

export function addString(data) {
  return {
    type: ADD_STRING,
    data,
  };
}

export function addStringSuccess(data) {
  return {
    type: ADD_STRING_SUCCESS,
    data,
  };
}

export function addStringError(error) {
  return {
    type: ADD_STRING_ERROR,
    error,
  };
}
