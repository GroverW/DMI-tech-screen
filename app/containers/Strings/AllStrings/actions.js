/*
 *
 * AllStrings actions
 *
 */

import {
  LOAD_STRINGS,
  LOAD_STRINGS_SUCCESS,
  LOAD_STRINGS_ERROR,
} from '../constants';

export function loadStrings(data) {
  return {
    type: LOAD_STRINGS,
    data,
  };
}

export function loadStringsSuccess(data) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    data,
  };
}

export function loadStringsError(error) {
  return {
    type: LOAD_STRINGS_ERROR,
    error,
  };
}
