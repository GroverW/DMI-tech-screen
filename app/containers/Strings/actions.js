/*
 *
 * Strings actions
 *
 */

import { RESET_STATUSES } from './constants';

export function resetStatuses() {
  return {
    type: RESET_STATUSES,
  };
}
