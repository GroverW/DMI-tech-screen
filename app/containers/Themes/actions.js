/*
 *
 * AllStrings actions
 *
 */

import { SET_THEME } from './constants';

export function setTheme(theme) {
  return {
    type: SET_THEME,
    theme,
  };
}
