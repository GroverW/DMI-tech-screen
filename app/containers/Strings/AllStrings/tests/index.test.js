/**
 *
 * Tests for AllStrings
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../../configureStore';
import 'jest-dom/extend-expect'; // add some helpful assertions
import { loadStrings } from '../actions';

import { AllStrings, mapDispatchToProps } from '../index';
import { DEFAULT_LOCALE } from '../../../../i18n';

describe('<AllStrings />', () => {
  let store;
  let mockTheme;

  beforeAll(() => {
    store = configureStore({}, browserHistory);

    mockTheme = {
      name: '',
      colors: {},
    };
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <AllStrings
            strings={[]}
            loading={false}
            errors={false}
            loaded={false}
            theme={mockTheme}
            loadStrings={loadStrings}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <AllStrings
            strings={[]}
            loading={false}
            errors={false}
            loaded={false}
            theme={mockTheme}
            loadStrings={loadStrings}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('Should load the body content', async () => {
    const { queryByText } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <AllStrings
            strings={[]}
            loading={false}
            errors={false}
            loaded={false}
            theme={mockTheme}
            loadStrings={loadStrings}
          />
        </IntlProvider>
      </Provider>,
    );

    expect(queryByText('Latest Strings')).toBeInTheDocument();
    expect(queryByText('Load More')).toBeInTheDocument();
  });

  describe('mapDispatchToProps', () => {
    describe('loadStrings', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);

        expect(result.loadStrings).toBeDefined();
      });

      it('should dispatch loadStrings when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const string = 'test';
        result.loadStrings(string);

        expect(dispatch).toHaveBeenCalledWith(loadStrings(string));
      });
    });
  });
});
