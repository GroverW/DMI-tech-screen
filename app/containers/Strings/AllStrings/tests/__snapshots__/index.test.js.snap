// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllStrings /> Should render and match the snapshot 1`] = `
<div>
  <h2>
    <span>
      Latest Strings
    </span>
  </h2>
  <div
    class="ContentContainer-sc-18efx5u-0 lpiSfa"
  >
    <button
      class="MoreButton-sc-78mmo7-0 kWezJA"
      type="button"
    >
      <span>
        Load More
      </span>
    </button>
  </div>
</div>
`;
